_Convert to App Engine Standard Project_ fails on missing WEB-INF directory

Using _Convert to App Engine Standard Project_ to convert a Springboot app, such as the `springboot-appengine-standard.zip` test project, may fail as there is no WEB-INF directory:

```
!ENTRY com.google.cloud.tools.eclipse.appengine.facets 4 0 2017-08-08 16:44:05.577
!MESSAGE Failed to convert project "springboot-appengine-standard".
!STACK 1
org.eclipse.wst.common.project.facet.core.FacetedProjectFrameworkException: Failed while installing App Engine Java Standard Environment JRE8.
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.callDelegate(FacetedProject.java:1507)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.modifyInternal(FacetedProject.java:441)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.mergeChangesInternal(FacetedProject.java:1181)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.access$2(FacetedProject.java:1117)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject$1.run(FacetedProject.java:324)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.modify(FacetedProject.java:339)
	at com.google.cloud.tools.eclipse.appengine.facets.FacetUtil.install(FacetUtil.java:240)
	at com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet.installAppEngineFacet(AppEngineStandardFacet.java:313)
	at com.google.cloud.tools.eclipse.appengine.facets.convert.AppEngineStandardProjectConvertJob.run(AppEngineStandardProjectConvertJob.java:50)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: org.eclipse.core.internal.resources.ResourceException: Resource '/springboot-appengine-standard/src/main/webapp/WEB-INF' does not exist.
	at org.eclipse.core.internal.resources.Resource.checkExists(Resource.java:335)
	at org.eclipse.core.internal.resources.Resource.checkAccessible(Resource.java:209)
	at org.eclipse.core.internal.resources.File.create(File.java:110)
	at org.eclipse.core.internal.resources.File.create(File.java:181)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallDelegate.createConfigFiles(StandardFacetInstallDelegate.java:152)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallDelegate.execute(StandardFacetInstallDelegate.java:51)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.callDelegate(FacetedProject.java:1477)
	... 11 more
!SUBENTRY 1 org.eclipse.wst.common.project.facet.core 4 0 2017-08-08 16:44:05.578
!MESSAGE Failed while installing App Engine Java Standard Environment JRE8.
!STACK 1
org.eclipse.core.internal.resources.ResourceException: Resource '/springboot-appengine-standard/src/main/webapp/WEB-INF' does not exist.
	at org.eclipse.core.internal.resources.Resource.checkExists(Resource.java:335)
	at org.eclipse.core.internal.resources.Resource.checkAccessible(Resource.java:209)
	at org.eclipse.core.internal.resources.File.create(File.java:110)
	at org.eclipse.core.internal.resources.File.create(File.java:181)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallDelegate.createConfigFiles(StandardFacetInstallDelegate.java:152)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallDelegate.execute(StandardFacetInstallDelegate.java:51)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.callDelegate(FacetedProject.java:1477)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.modifyInternal(FacetedProject.java:441)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.mergeChangesInternal(FacetedProject.java:1181)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.access$2(FacetedProject.java:1117)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject$1.run(FacetedProject.java:324)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.modify(FacetedProject.java:339)
	at com.google.cloud.tools.eclipse.appengine.facets.FacetUtil.install(FacetUtil.java:240)
	at com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet.installAppEngineFacet(AppEngineStandardFacet.java:313)
	at com.google.cloud.tools.eclipse.appengine.facets.convert.AppEngineStandardProjectConvertJob.run(AppEngineStandardProjectConvertJob.java:50)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
!SUBENTRY 2 org.eclipse.core.resources 4 368 2017-08-08 16:44:05.578
!MESSAGE Resource '/springboot-appengine-standard/src/main/webapp/WEB-INF' does not exist.
```

--- a/plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java
@@ -148,6 +148,9 @@ public class StandardFacetInstallDelegate extends AppEngineFacetInstallDelegate 
     if (appEngineWebXml.exists()) {
       return;
     }
+    if (!webInfDir.exists()) {
+      ResourceUtils.createFolders(webInfDir, progress.newChild(1));
+    }
 
     appEngineWebXml.create(new ByteArrayInputStream(new byte[0]), true, progress.newChild(2));
     String configFileLocation = appEngineWebXml.getLocation().toString();

--- a/plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java
@@ -134,24 +134,18 @@ public class StandardFacetInstallDelegate extends AppEngineFacetInstallDelegate 
     // The virtual component model is very flexible, but we assume that
     // the WEB-INF/appengine-web.xml isn't a virtual file remapped elsewhere
     IFolder webInfDir = WebProjectUtil.getWebInfDirectory(project);
-
     if (webInfDir == null) {
       webInfDir =
           project.getFolder(WebProjectUtil.DEFAULT_WEB_PATH).getFolder(WebProjectUtil.WEB_INF);
-      ResourceUtils.createFolders(webInfDir, progress.newChild(3));
     }
-
     progress.worked(1);
 
     IFile appEngineWebXml = webInfDir.getFile(APPENGINE_WEB_XML);
-
     if (appEngineWebXml.exists()) {
       return;
     }
-    if (!webInfDir.exists()) {
-      ResourceUtils.createFolders(webInfDir, progress.newChild(1));
-    }
 
+    ResourceUtils.createFolders(webInfDir, progress.newChild(1));
     appEngineWebXml.create(new ByteArrayInputStream(new byte[0]), true, progress.newChild(2));
     String configFileLocation = appEngineWebXml.getLocation().toString();
     Templates.createFileContent(

